From cc49d15554ecfa095ecc0350f3d816f7713e94c9 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 15 Mar 2014 14:01:40 -0500
Subject: [PATCH 273/390] Remove CONFIG_80211D

We always build in 802.11d capability. A module parameter can disable it.

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 core/rtw_mlme_ext.c    | 6 ------
 include/autoconf.h     | 2 --
 include/drv_types.h    | 5 -----
 include/rtw_mlme_ext.h | 3 ---
 os_dep/os_intfs.c      | 6 ------
 5 files changed, 22 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_mlme_ext.c
+++ b/drivers/staging/rtl8192du/core/rtw_mlme_ext.c
@@ -8888,7 +8888,6 @@ unsigned int receive_disconnect(struct r
 	return _SUCCESS;
 }
 
-#ifdef CONFIG_80211D
 static void process_80211d(struct rtw_adapter *adapt,
 			   struct wlan_bssid_ex *bssid)
 {
@@ -9015,7 +9014,6 @@ static void process_80211d(struct rtw_ad
 			       (chplan_sta[i].ChannelNum <= 14)) {
 				chplan_new[k].ChannelNum =
 				    chplan_sta[i].ChannelNum;
-/*				chplan_new[k].ScanType = chplan_sta[i].ScanType; */
 				chplan_new[k].ScanType = SCAN_PASSIVE;
 				i++;
 				k++;
@@ -9084,7 +9082,6 @@ static void process_80211d(struct rtw_ad
 			while ((i < MAX_CHANNEL_NUM) && (chplan_sta[i].ChannelNum != 0)) {
 				chplan_new[k].ChannelNum =
 				    chplan_sta[i].ChannelNum;
-/*				chplan_new[k].ScanType = chplan_sta[i].ScanType; */
 				chplan_new[k].ScanType = SCAN_PASSIVE;
 				i++;
 				k++;
@@ -9142,7 +9139,6 @@ static void process_80211d(struct rtw_ad
 		i++;
 	}
 }
-#endif
 
 /****************************************************************************
 
@@ -9204,9 +9200,7 @@ void report_survey_event(struct rtw_adap
 		kfree(pevtcmd);
 		return;
 	}
-#ifdef CONFIG_80211D
 	process_80211d(adapt, &psurvey_evt->bss);
-#endif
 
 	rtw_enqueue_cmd(pcmdpriv, pcmd_obj);
 
--- a/drivers/staging/rtl8192du/include/autoconf.h
+++ b/drivers/staging/rtl8192du/include/autoconf.h
@@ -56,8 +56,6 @@
 	#define CONFIG_TSF_RESET_OFFLOAD 1			/*  For 2 PORT TSF SYNC. */
 #endif	/*  CONFIG_CONCURRENT_MODE */
 
-#define CONFIG_80211D
-
 /* Interface  Related Config */
 
 #define CONFIG_PREALLOC_RECV_SKB	1
--- a/drivers/staging/rtl8192du/include/drv_types.h
+++ b/drivers/staging/rtl8192du/include/drv_types.h
@@ -149,14 +149,9 @@ struct registry_priv {
 
 	u8  special_rf_path; /* 0: 2T2R ,1: only turn on path A 1T1R, 2: only turn on path B 1T1R */
 	u8	mac_phy_mode; /* 0:by efuse, 1:smsp, 2:dmdp, 3:dmsp. */
-
-#ifdef CONFIG_80211D
 	u8 enable80211d;
-#endif
-
 	u8 ifname[16];
 	u8 if2name[16];
-
 	u8 notch_filter;
 
 #ifdef CONFIG_MULTI_VIR_IFACES
--- a/drivers/staging/rtl8192du/include/rtw_mlme_ext.h
+++ b/drivers/staging/rtl8192du/include/rtw_mlme_ext.h
@@ -412,10 +412,7 @@ struct mlme_ext_priv
 	/* recv_decache check for Action_public frame */
         u8 action_public_dialog_token;
 	u16	 action_public_rxseq;
-
-#ifdef CONFIG_80211D
 	u8 update_channel_plan_by_ap_done;
-#endif
 	u8 active_keep_alive_check;
 };
 
--- a/drivers/staging/rtl8192du/os_dep/os_intfs.c
+++ b/drivers/staging/rtl8192du/os_dep/os_intfs.c
@@ -131,9 +131,7 @@ int rtw_mc2u_disable = 0;
 
 static int rtw_mac_phy_mode; /* 0:by efuse, 1:smsp, 2:dmdp, 3:dmsp. */
 
-#ifdef CONFIG_80211D
 static int rtw_80211d = 0;
-#endif
 
 static char *ifname = "wlan%d";
 module_param(ifname, charp, 0644);
@@ -193,9 +191,7 @@ module_param(rtw_mc2u_disable, int, 0644
 
 module_param(rtw_mac_phy_mode, int, 0644);
 
-#ifdef CONFIG_80211D
 module_param(rtw_80211d, int, 0644);
-#endif
 
 static uint rtw_notch_filter = RTW_NOTCH_FILTER;
 module_param(rtw_notch_filter, uint, 0644);
@@ -283,9 +279,7 @@ static uint loadparam(struct rtw_adapter
 	registry_par->mac_phy_mode = rtw_mac_phy_mode;
 	GlobalDebugLevel = debug;
 
-#ifdef CONFIG_80211D
 	registry_par->enable80211d = (u8)rtw_80211d;
-#endif
 
 	snprintf(registry_par->ifname, 16, "%s", ifname);
 	snprintf(registry_par->if2name, 16, "%s", if2name);
